File: Data Structures & Algorithms/climbing-stairs/submission-2.java
class Solution {
    public int climbStairs(int n) {
        //Base Case -> start computing distinct ways to climb top
        if (n==0 || n==1) return 1;
        //Example:2 -> climbStairs(1) = 1 + climbStairs(0) = 1. Output = 1+1 = 2
        return climbStairs(n-1) + climbStairs(n-2);
    }
}

File: Data Structures & Algorithms/remove-element/submission-2.cpp
class Solution {
public:
    int removeElement(vector<int>& nums, int val) {
        /*
        Input: nums = [3,2,2,3], val = 3
        Output: k = 2, nums = [2,2,_,_]
        */
        int i = 0, n = nums.size();
        while (i < n) {
            if (nums[i] == val) nums[i] = nums[--n];
            else i++;
        }
        return n;
    }
};